app/react-native/src/preview/components/StoryView/index.tsx

```diff
@@ -75,41 +59,19 @@ export default class StoryView extends Component<Props, State> {
   );
 
   render() {
-    const { listenToEvents } = this.props;
+    const {onDevice, stories} = this.props;
 
-    if (listenToEvents) {
-      return this.renderListening();
-    } else {
-      return this.renderOnDevice();
-    }
-  }
+    const selection = stories.getSelection();
 
-  renderListening = () => {
-    if (!this.state) {
-      return null;
-    }
-    const { storyFn, selection } = this.state;
-    const { kind, story } = selection;
+    const {kind, story, storyFn} = selection;
 
     return storyFn ? (
-      <View key={`${kind}:::${story}`} style={{ flex: 1 }}>
+      <View key={`${kind}:::${story}`} style={{flex: 1}}>
```

Comment:
does this still work

Reply:
@benoitdion, what exactly do you mean? I will change it to story.id since that will be shorter.